### ISSUE TITLE
[Python-pachyderm] Check validity of code snippets in doc + Add note about which python-pachyderm version to import depending on Pachyderm release + Test validity of examples

### ISSUE THREAD

**Issue by nadegepepin** [pachyderm/pachyderm#6155]
I have identified the pages in our general documentation and examples where python-pachyderm is mentioned/explained.

- https://docs.pachyderm.com/latest/how-tos/use-pachyderm-ide/
- http://docs.pachyderm.com/latest/how-tos/use-pachyderm-ide/using-pachyderm-ide/
- http://docs.pachyderm.com/latest/how-tos/developer-workflow/
- http://docs.pachyderm.com/latest/reference/clients/

Our Spout 101 example uses python-pachyderm in: https://github.com/pachyderm/pachyderm/blob/master/examples/spouts/spout101/src/consumer/main.py

Additionally, our python-pachyderm repo has its own set of examples:
https://github.com/pachyderm/python-pachyderm/tree/master/examples

A couple of questions related to ^^:

- I am not entirely sure that the snippets of code that are displayed in the doc are still valid: **Check the validity of code snippets in doc.** 
- Regarding the examples directory in pachyderm/python-pachyderm repo: **Make sure that those examples still run properly.**
- Corollary, the `create-python-pipeline` seems to be an ancestor of build pipelines: ** Check if it still works with 1.x and 2.0.**
- How do we know, in https://pypi.org/project/python-pachyderm, which version of python-pachyderm is compatible with each version of pachyderm? : **Add a note in the doc to specify which version of python-pachyderm to use depending on the version of Pachyderm you are on.**

**Comment by nadegepepin** [comment]
- The python team took a pass at all examples
- python-pachyderm v6 (1.13.x and before) and v7 (2.0.x) has been released on RTD and their links updated in the documentation
- In 2.0 documentation - The previous **How-tos Use Pachyderm IDE** section went through a clean-up.
There is now _one simple page_ that explains what Notebooks are, the base image of a notebook, a link to the main tutorial, and a mention of all the available examples. Hub is linked as the easiest way to get started.
I removed anything python-pachyderm from Notebooks. python-pachyderm and Notebooks are two separate matters. I have, however, made sure to list and link all that is available in our custom image, including pachctl and python-pachyderm.
- Python-pachyderm is mentioned in the list of clients that Pachyderm maintains (with links to the installation, API reference docs, and examples), along with node-pachyderm and the Go client.  I made sure to point users to the correct version of the client, depending on their version of Pachyderm.
